airflow/providers/common/sql/hooks/sql.py

```diff
@@ -207,6 +208,17 @@ def get_uri(self) -> str:
         conn.schema = self.__schema or conn.schema
         return conn.get_uri()
 
+    @property
+    def sqlalchemy_url(self) -> URL:
+        """
+        Return a Sqlalchemy.engine.URL object from the connection.
+
+        Needs to be implemented in the provider subclass to return the sqlalchemy.engine.URL object.
+
+        :return: the extracted sqlalchemy.engine.URL object.
+        """
+        raise NotImplementedError("sqlalchemy_url property should be implemented in the provider subclass.")
```

Comment:
```suggestion
        qualname = f"{self.__class__.__module__}.{self.__class__.__qualname__}"
        if qualname != "airflow.providers.common.sql.hooks.sql.DbApiHook":
            msg = f"{qualname!r} not implements/supports built SQLAlchemy URI."
        else:
            msg = "`sqlalchemy_url` property should be implemented in the provider subclass."
        raise NotImplementedError(msg)
```

Let's a bit beautify output